develop/application-devs/smart-contracts/smart-contracts.md

```diff
@@ -0,0 +1,165 @@
+
+## Developing a Smart Contract Versus a Parachain
+
+!!!info "For a more technical/thorough breakdown, refer to the [`polkadot_sdk_docs`](https://paritytech.github.io/polkadot-sdk/master/polkadot_sdk_docs/reference_docs/runtime_vs_smart_contract/index.html){target=\_blank}"
+
+[When should one build a Substrate (Polkadot SDK) runtime versus a Substrate (Polkadot SDK) smart contract](https://stackoverflow.com/a/56041305){target=\_blank}?
+This post also answers the question more technically of when a developer might choose to develop a runtime versus a smart contract.
+
+### Layer of Abstraction
+
+When you write a smart contract, you are creating the instructions that associate with and deploy on
+a specific chain address.
+
+In comparison, a runtime module on a parachain is the entire logic of a chain's state transitions
+(what's called a state transition function).
+
+Smart contracts must consciously implement a path to upgrading in the future, while parachains have the ability to swap
+out their code through a forkless upgrade.
+
+When you build a smart contract, it will eventually be deployed to a target chain with its own
+environment. Parachains allow the developer to declare the environment of their own chain, even
+allowing others to write smart contracts for it.
+
+The concept of *gas*, or *fees* in general, also are approached differently between the two. Smart contracts are bound by the gas-metering model, whereas runtimes (and their subsequent modules) are much more flexible in terms of the fee models that can be employed.
+
+## Building a Smart Contract
+
+The relay chain doesn't natively support smart contracts. However, since the parachains that connect to can support arbitrary state transitions, they support smart contracts.
+
+The Polkadot SDK presently supports smart contracts out-of-the-box in several ways:
+
+- The EVM pallet offered by [Frontier](https://github.com/paritytech/frontier){target=\_blank}
+- The [contracts pallet](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/frame/contracts/){target=\_blank} in the FRAME library for Wasm-based contracts
+
+### Frontier EVM Contracts
+
+[Frontier](https://github.com/paritytech/frontier) is the suite of tools that enables a Substrate chain to run Ethereum contracts (EVM) natively with the same API/RPC interface, Ethereum exposes on Substrate. Ethereum Addresses can also be mapped directly to and from Substrate's SS58 scheme from existing accounts.
+
+- [Pallet EVM](https://docs.rs/pallet-evm/latest/pallet_evm/){target=\_blank} - enables functionality of running EVM (Solidity) contracts
+- [Pallet Ethereum](https://docs.rs/pallet-ethereum/latest/pallet_ethereum/){target=\_blank} - Ethereum RPC implementation, block emulation, and Ethereum transaction validation
+
+### Contracts Pallet
+
+The contracts pallet (`pallet_contracts`) implements a WebAssembly based approach to smart contracts.
```

Comment:
I'd add a link to the contracts pallet source code :)

Reply:
Also, I'd add a bit more of intro here, just because it seems a bit unconnected to start talking about pallet contracts features directly immediately after this sentence